syntax = "proto3";

import "shiva/core/communication_objects/enums.proto";

package communication_objects;

message AgentMetricProto {
    float steps_per_episode = 1;
    float step_count = 2;
    float temp_done_counter = 3;
    float done_count = 4;
    float reward_per_step = 5;
    float reward_per_episode = 6;
    float reward_total = 7;
}

message TrainingMetricsProto {
    map <string, AgentMetricProto> data = 1;
}

message EvaluationMetricsProto {
    map <string, AgentMetricProto> data = 1;
}

// Message used for components to Check-In and send initial parameters in between modules
message MetricsProto {
    int32 id = 1;
    ComponentType type = 2;
    StatusType status = 3;
    string address = 4;
    oneof data {
        AgentMetricProto agent = 5;
        TrainingMetricsProto train = 6;
        EvaluationMetricsProto eval = 7;
    }
    string extra = 8;
}